diagnostics/tropical_pacific_sea_level/tropical_pacific_sea_level.py

```diff
@@ -0,0 +1,728 @@
+
+        # store all model data
+        ds_model_list[var] = da_model
+
+        # calculate mean
+        mean_list[var] = ds_model_list[var].mean(dim='time').compute()
+        ds_model_list[var] = ds_model_list[var]-mean_list[var]
+
+        # calculate seasonality
+        season_list[var] = ds_model_list[var].groupby('time.month').mean(dim='time').compute()
+        ds_model_list[var] = ds_model_list[var].groupby('time.month')-season_list[var]
+
+        # remove linear trend
+        linear_list[var] = da_linregress(ds_model_list[var],stTconfint=0.99)
+
+    linear_mlist[model] = linear_list
+    mean_mlist[model] = mean_list
+    season_mlist[model] = season_list
+    ds_model_mlist[model] = ds_model_list
+
+#
+# # detrend
+# for nmodel,model in enumerate(Model_name):
+#     for nvar,var in enumerate(Model_varname):
+#         da_time = ds_model_mlist[model][var].time.copy()
+#         year = ds_model_mlist[model][var]['time.year'].values
+#         month = ds_model_mlist[model][var]['time.month'].values
+#         da_time.values = year+month/12.
+#         ds_model_mlist[model][var] = ds_model_mlist[model][var]-                                     (da_time*linear_mlist[model][var]['slope']+linear_mlist[model][var]['intercept'])
+
+
+# # Observation
+# constant setting
+obs_year_range = [[1950,2011],[1993,2018,9]]
+Obs_varname = [['tx','ty'],['adt']]
+Obs_name = ['WASwind','CMEMS']
+
+# inputs
+obsin = {}
+obspath = {}
+
+obs = Obs_name[0]
+obsdir = str(os.getenv('OBS_DATA'))
+obsfile = [['waswind_v1_0_1.monthly.nc'],['waswind_v1_0_1.monthly.nc']]
+obspath[obs]=[obsdir,obsfile]
+
+obs = Obs_name[1]
+obsdir = str(os.getenv('OBS_DATA'))
+obsfile = [['dt_global_allsat_phy_l4_monthly_adt.nc']]
+obspath[obs]=[obsdir,obsfile]
+
+
+for nobs,obs in enumerate(Obs_name):
+    obsdir = obspath[obs][0]
+    obsfile = obspath[obs][1]
+    multivar = []
+    for file in obsfile :
+        if len(file) == 1 :
+            multivar.append([os.path.join(obsdir,file[0])])
+        elif len(file) > 1 :
+            multifile = []
+            for ff in file :
+                multifile.append(os.path.join(obsdir,ff))
+            multivar.append(multifile)
+    obsin[obs] = multivar
+
+
+# initialization of dict and list
+ds_obs_mlist = {}
+obs_mean_mlist = {}
+obs_season_mlist = {}
+obs_linear_mlist = {}
+
+for nobs,obs in enumerate(Obs_name):
+    ds_obs_list = {}
+    obs_mean_list = {}
+    obs_season_list = {}
+    obs_linear_list = {}
+    for nvar,var in enumerate(Obs_varname[nobs]):
+        print('read %s %s'%(obs,var))
+
+        # read input data
+        #-- single file
+        if len(obsin[obs][nvar]) == 1 :
+
+            # find out dimension name
+            da = xr.open_dataset(obsin[obs][nvar][0])
+            obsdims = list(da[var].dims)
+
+#             ds_obs = xr.open_dataset(obsin[obs][nvar][0],chunks={obsdims[0]:50,obsdims[1]:50,obsdims[2]:50},use_cftime=True)
```

Comment:
If there is a reason to retain this line, leave a comment (e.g., "uncomment for debugging"); otherwise delete it.

Reply:
Agreed. Will remove the commented part.